### PR TITLE
Re-implement some modules in C

### PR DESCRIPTION
**Pull request by yichoi**
Buffer, Console, Constant, HttpParser, Process, Testdriver, Timer

Related issue: #439

### CODE REVIEW

File: src/module/iotjs_module_buffer.c [hunk 1]
```diff
@@ -0,0 +1,374 @@
...
+                               jbuiltin,
+                               (uintptr_t)bufferwrap,
+                               (JFreeHandlerType)iotjs_bufferwrap_destroy);
+  if (length > 0) {
+    _this->buffer = iotjs_buffer_allocate(length);
+    IOTJS_ASSERT(_this->buffer != NULL);
+  } else {
+    _this->buffer = NULL;
+  }
+  _this->length = length;
```
Comment (nova0821) [inline src/module/iotjs_module_buffer.c]: IMHO, if length is lees than 0, ``_this->length`` will get a negative value. Don't we need to check it?

File: src/module/iotjs_module_httpparser.c [hunk 2]
```diff
@@ -0,0 +1,554 @@
...
+//http-parser callbacks
+HTTP_CB( on_message_begin );
+HTTP_DATA_CB( on_url );
+HTTP_DATA_CB( on_status );
+HTTP_DATA_CB( on_header_field );
+HTTP_DATA_CB( on_header_value );
+HTTP_CB( on_headers_complete );
+HTTP_DATA_CB( on_body );
+HTTP_CB( on_message_complete );
+
```
Comment (nova0821) [inline src/module/iotjs_module_httpparser.c]: Please check coding style guideline. We have a rule - ``Do not insert spaces after open paren and before close paren.``

File: src/module/iotjs_module_httpparser.c [hunk 3]
```diff
@@ -0,0 +1,554 @@
...
+  if (!nativeparser->upgrade && nparsed != buf_len) {
+    // nparsed should equal to buf_len except UPGRADE protocol
+    enum http_errno err = HTTP_PARSER_ERRNO(nativeparser);
+    iotjs_jval_t eobj = iotjs_jval_create_error("Parse Error");
+    iotjs_jval_set_property_number(&eobj, "byteParsed", 0);
+    iotjs_jval_set_property_string_raw(&eobj, "code", http_errno_name(err));
+    iotjs_jhandler_return_jval(jhandler, &eobj);
+    iotjs_jval_destroy(&eobj);
+  }
+  else{
```
Comment (nova0821) [inline src/module/iotjs_module_httpparser.c]: Please check coding style in this line. - ``The else goes on the same line as the closing brace.``

**Comment by yichoi** [comment]
fixed. thanks @nova0821

**Comment by zherczeg** [comment]
LGTM

**Pull request merged**